pkgs/development/compilers/julia/lts.nix

```diff
@@ -0,0 +1,148 @@
+{ fetchzip, fetchFromGitHub, lib, pkgs, stdenv
+# Build tools
+, cmake, perl, python3, which
+# Dependencies
+, arpack, curl, blas, gmp, lapack, libssh2, libunwind, mpfr, pcre2
+# Testing
+, openssl
+# Darwin
+, CoreServices, ApplicationServices }:
+
+# Ensure dependencies from Nixpkgs are equal or greater to vendored versions.
+assert lib.versionAtLeast (lib.strings.getVersion gmp) "6.1.2";
+assert lib.versionAtLeast arpack.version "3.3.0";
+assert lib.versionAtLeast curl.version "7.56.0";
+assert lib.versionAtLeast libssh2.version "1.8.2";
+assert lib.versionAtLeast libunwind.version "1.1";
+assert lib.versionAtLeast mpfr.version "4.0.2";
+assert lib.versionAtLeast pcre2.version "10.30";
+# Can not assert for BLAS and LAPACK due to multiple providers, Julia uses:
+#   * OpenBLAS: e8a68ef261a33568b0f0cf53e0e2287e9f12e69e
+#   * LAPACK: 3.5.0
+
+# Ensure BLAS and Lapack integer size compatibility.
+assert (blas.isILP64 == lapack.isILP64);
+
+let
+  # Pre-1.6.x Julia is not compatible with libgit2 1.x, use an earlier version.
+  libgit2 = pkgs.libgit2.overrideAttrs (oldAttrs: rec {
+    version = "0.28.5";
+    src = fetchFromGitHub {
+      owner = "libgit2";
+      repo = "libgit2";
+      rev = "v${version}";
+      sha256 = "1i26l4annj07p5draj496niwh1mdba7ymr657w09m5dlw92bk1wn";
+    };
+  });
+
+in stdenv.mkDerivation rec {
+  pname = "julia";
+  version = "1.0.5";
+  src = fetchzip {
+    url = "https://github.com/JuliaLang/${pname}/releases/download/v${version}/${pname}-${version}-full.tar.gz";
+    sha256 = "0m9c43y6ma7vgzk3rhdn60ll2y6gi3vd0pba79nma4028l5ki0k4";
+  };
+
+  patches = [
+    # Replace with patch based on PR below once merged:
+    #   https://github.com/JuliaLang/julia/pull/40178
+    ./julia-lts-libgit2-0.28.5-support.patch
+    ./julia-lts-skip-tests-requiring-network.patch
+  ];
+  prePatch = ''
+    # Disable Distributed tests as they fail in the sandbox.
+    echo > stdlib/Distributed/test/runtests.jl
+  '';
+
+  strictDeps = true;
+  nativeBuildInputs = [ cmake perl python3 which ];
+  buildInputs = [
+    arpack
+    blas
+    curl
+    gmp
+    lapack
+    libgit2
+    libssh2
+    libunwind
+    mpfr
+    pcre2
+  ] ++ lib.optionals stdenv.isDarwin [ CoreServices ApplicationServices ];
+  # Julia creates symlinks in `lib/julia` by searching `LD_LIBRARY_PATH`.
+  LD_LIBRARY_PATH = lib.makeLibraryPath buildInputs;
+  # CMake is used to build the vendored dependencies, not Julia itself.
+  dontUseCmakeConfigure = true;
+  enableParallelBuilding = true;
```

Comment:
```suggestion

```

set by cmake.

Reply:
See response for `stable.nix` for the same line.